enginetest/queries.go

```diff
@@ -615,6 +615,27 @@ var QueryTests = []QueryTest{
 			{sql.MustJSON(`{ "a": [5, 3], "d": 6, "e": 2}`)},
 		},
 	},
+	{
+		Query: `SELECT JSON_ARRAY('{"b": 2, "a": [1, 8], "c": null}', null, 4, '[true, false]', "do")`,
+		Expected: []sql.Row{
+			{sql.MustJSON(`["{\"b\": 2, \"a\": [1, 8], \"c\": null}", null, 4, "[true, false]", "do"]`)},
+		},
+	},
+	{
+		Query: `SELECT JSON_ARRAY('a', 1, datetime('2020-01-01T12:00:00'))`,
+		Expected: []sql.Row{{
+			sql.MustJSON(`["a", 1, "2020-01-01 12:00:00 +0000 UTC"]`)},
+		},
+	},
+	{
+		Query: `SELECT JSON_ARRAY(pk, c1, c2, c3) FROM jsontable`,
```

Comment:
Add test for empty set. Add additional tests that compose JSON_ARRAY with additional JSON functions to ensure type validity

Reply:
You should also exercise this function outside of a table.